### ISSUE TITLE
proxying openvpn stopped working?

### ISSUE THREAD

**Issue by Bretos** [yrutschle/sslh#80]
Hello guys! I am running openvpn server version 2.3.10 and client version 2.3.9. sslh is running on Debian 7, sslh version is 1.17.

I've launched sslh with following options:

> sslh --listen 0.0.0.0 443 --openvpn 127.0.0.1 1194 --ssh 127.0.0.1 22 --ssl 127.0.0.2 4443 --xmpp 10.0.0.13 5222

Proxying for ssh, ssl, xmpp works. For openvpn it doesn't!

All that client logs say is:

```
mar 02 10:14:15 localhost openvpn@Dell[20363]: TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
mar 02 10:14:15 localhost openvpn@Dell[20363]: TLS Error: TLS handshake failed
```

openvpn server do not seem to get any trial of connection to.
openvpn server has no 'local' directive.

I forgot to mention that without sslh in the middle openvpn is working perfectly.

**Comment by yrutschle** [comment]
Hi,

The OpenVPN probe in sslh hasn't changed since 2013 / v1.14
(I think). Did you upgrade OpenVPN recently? If so, did it
work before?

(This is the second OpenVPN issue in two days, so now I
wonder if there has been a change in OpenVPN...)

Y.

**Comment by Bretos** [comment]
openvpn has been updated to "openvpn_2.3.10" on 21st of January this year. 

I don't know if sslh worked before this upgrade or not, since it is just recently that I tried to use it.

**Comment by delcara** [comment]
Did you ever resolve this issue? I have the same issue except that I use transparent mode in SSLH. Everything is working except OpenVPN, i have the same error: 

```
Wed Jul 25 09:28:31 2018 TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
Wed Jul 25 09:28:31 2018 TLS Error: TLS handshake failed
```

**Comment by ShaneMalachow** [comment]
Similar error happening over here as well. Getting this working would be really helpful. 
```
Mon Oct 29 16:22:31 2018 TCP/UDP: Preserving recently used remote address: [AF_INET]xxx.xxx.xxx.xxx:8443
Mon Oct 29 16:22:31 2018 UDP link local (bound): [AF_INET][undef]:1194
Mon Oct 29 16:22:31 2018 UDP link remote: [AF_INET]xxx.xxx.xxx.xxx:8443
Mon Oct 29 16:23:31 2018 TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
Mon Oct 29 16:23:31 2018 TLS Error: TLS handshake failed
```

**Comment by yrutschle** [comment]
I'm confused: OpenVPN receives the connection from sslh, then the TLS handshake times out?

Also, what's that message about UDP? sslh does nothing about UDP, you need the connection to occur entirely in TCP.

**Comment by yrutschle** [comment]
I had a look at traffic coming from a recent client, and it looks like the connection is now established using TLS. Using the current TLS probe (possibly specifying SNI, which the client sets correctly) works correctly. I don't know (yet) if that's enough to establish the connection, as I don't have a working server.

**Comment by yrutschle** [comment]
Ok, I tested with OpenVPN 2.4.0 (client and server) and the Android client 3.0.5. Running sslh with -v I see the first packet as:
```
0x000000: 00 0e 38 8b 56 e8 65 b5 98 5c d9 00 00 00 00 00 ..8.V.e..\......
```
(with some variations). The current probe works fine and I can establish a VPN, both from Linux and from Android (ignore what I said earlier about TLS -- I'm not sure how I saw that).

What does the first packet look like for you?
